Competition2017/src/competition/subsystems/fuel_launcher/FuelLauncherSubsystem.java

```diff
@@ -0,0 +1,185 @@
+        // I call this Ticks per Centisecond, or TPC
+        shooterSpeedThresh = propManager.createPersistentProperty("FuelLauncher nominal speed thresh (TPC)", 1);
+        shooterCurrentSpeed = propManager.createEphemeralProperty("FuelLauncher Current Speed (TPC)", 0);
+        shooterTargetSpeed = propManager.createEphemeralProperty("FuelLauncher Goal Speed (TPC)", 0);
+        shooterOutputPower = propManager.createEphemeralProperty("FuelLauncher Voltage", 0);
+        atSpeedProp = propManager.createEphemeralProperty("Is FuelLauncher at speed?", false);
+        shooterTalonError = propManager.createEphemeralProperty("FuelLauncher Speed error", 0);
+        
+        this.masterMotor = factory.getCANTalonSpeedController(1);
+        //this.slaveMotor = factory.getCANTalonSpeedController(8);
+        initiallyConfigureMotorTeam(masterMotor);
+        masterMotor.createTelemetryProperties("FuelLauncher master", propManager);
+        //slaveMotor.createTelemetryProperties("FuelLauncher slave", propManager);
+    }
+
+    private void initiallyConfigureMotorTeam(XCANTalon master) {
+        // TODO: Check faults and voltage/temp/current
+        log.info("Configuring motor");
+        // Master config
+        master.setFeedbackDevice(FeedbackDevice.QuadEncoder);
+        //master.setBrakeEnableDuringNeutral(false);
+        // TODO: move inversion
+                
+        master.reverseSensor(true);
+        master.setInverted(true);
+        //master.enableLimitSwitches(false, false);
+        log.info("Configuring voltages");
+        master.configNominalOutputVoltage(0,  -0);
+        master.configPeakOutputVoltage(12, -12);
+
+        master.setProfile(0);
+        master.setF(.099);
+        updateMotorConfig(master);
+        master.setControlMode(TalonControlMode.Speed);
+        //master.setControlMode(TalonControlMode.PercentVbus);
+
+        master.set(0);
+        
+        // Slave config
+        /*slave.configNominalOutputVoltage(0,  -0);
+        slave.configPeakOutputVoltage(12, -12);
+        slave.enableLimitSwitches(false, false);
+        
+        slave.setControlMode(TalonControlMode.Follower);
+        slave.set(master.getDeviceID());*/
+    }
+    
+    private void updateMotorConfig(XCANTalon motor) {
+        motor.setP(p.get());
+        motor.setI(i.get());
+        motor.setD(d.get());
+        motor.setF(f.get());
+    }
+    
+    /**
+     * Sets the output power of the shooter directly - no PID of any kind
+     * @param power
+     */
+    public void setLauncherPower(double power) {
+        masterMotor.ensureTalonControlMode(TalonControlMode.PercentVbus);
+        masterMotor.set(power);
+    }
+    
+    /**
+     * Returns the last set motor power. Only works if the motor is currently in PercentVbus mode - otherwise, returns 0.
+     * @return
+     */
+    public double getLauncherPower() {
+        if (masterMotor.getControlMode() != TalonControlMode.PercentVbus) {
```

Comment:
This default behavior is unexpected. Why not return the power being sent regardless of the mode?